client/src/components/WorkflowInvocationState/WorkflowInvocationBiocompute.vue

```diff
@@ -0,0 +1,140 @@
+                                    class="form-control"
+                                    id="fetch"
+                                    placeholder="https://beta.portal.aws.biochemistry.gwu.edu"
+                                    v-model="form.fetch"
+                                />
+                            </div>
+                            <div class="form-group">
+                                <label for="authorization">User API Key</label>
+                                <input
+                                    type="text"
+                                    class="form-control"
+                                    id="authorization"
+                                    v-model="form.authorization"
+                                />
+                            </div>
+                            <div class="form-group">
+                                <label for="table">Table</label>
+                                <input
+                                    type="text"
+                                    class="form-control"
+                                    id="table"
+                                    placeholder="bco_draft"
+                                    v-model="form.table"
+                                />
+                            </div>
+                            <div class="form-group">
+                                <label for="owner_group">Group</label>
+                                <input
+                                    type="text"
+                                    class="form-control"
+                                    id="owner_group"
+                                    placeholder="bco_drafters"
+                                    v-model="form.owner_group"
+                                />
+                            </div>
+                            <div class="form-group">
+                                <button class="btn btn-primary">Submit</button>
+                            </div>
+                        </form>
+                    </div>
+                </details>
+            </div>
+            <div>
+                <a class="bco-json" :href="bcoJSON"><b>Download BioCompute Object</b></a>
+            </div>
+        </details>
+    </div>
+</template>
+<script>
+import { getRootFromIndexLink } from "onload";
+import { mapGetters } from "vuex";
+import axios from "axios";
+
+const getUrl = (path) => getRootFromIndexLink() + path;
+
+export default {
+    props: {
+        invocation: {
+            required: true,
+            type: String,
+        },
+    },
+    data() {
+        return {
+            bco: {
+                type: Object,
+                required: true,
+            },
+            form: {
+                fetch: "",
+                authorization: "",
+                table: "",
+                owner_group: "",
+            },
+        };
+    },
+    created: function () {
```

Comment:
<!-- hypothetical data wrapper around your form, separates data manipulation from data rendering -->

<template>
    <!-- this form would only render the inputs now, not also be in charge of the concern of
    persisting the data -->
    <WorkflowInvocationBioCompute v-if="bco" v-model="bco" />
</template>

<script>
// Importing your ajax calls as separate functions will make it easier to unit test this component
import { saveInvocation, getInvocation } from "./queries";

// Store your bco temp data exterior to the form rendering
export default {
    props: {
        invocation: { required: true },
    },
    data() {
        bco: null 
    },
    watch: {
        bco(newVal, oldVal) {
            if (newVal !== oldVal) {
                // do something with the form edit results
                this.save(newVal);
            }
        }
    },
    created() {
        this.load();     
    },
    methods: {
        async load() {
            // Do your ajax load here and set to 
            this.bco = await getInvocation(this.invocation);
        },
        async save(newVal) {
            // basically the same as your submitForm method
            return await saveInvocation(newVal);
        }
    }
}

</script>

Reply:
@Nerdinacan 
I am not sure I fully understand the implementation of your comment here. Are you suggesting I create another component with the structure you outlined above?

Reply:
@nsoranzo suggested integrating the BCO API key into the Galaxy  user preferences